server/blob/blob-union/src/main/java/org/apache/james/blob/union/UnionBlobStore.java

```diff
@@ -105,6 +106,21 @@ private UnionBlobStore(BlobStore currentBlobStore, BlobStore legacyBlobStore) {
         }
     }
 
+    @Override
+    public BucketName getDefaultBucketName() {
+        Preconditions.checkState(
+            currentBlobStore.getDefaultBucketName()
+                .equals(legacyBlobStore.getDefaultBucketName()),
+            "currentBlobStore and legacyBlobStore doen't have same defaultBucketName which could lead to " +
+                "unexpected result when interact with other APIs");
+        try {
+            return currentBlobStore.getDefaultBucketName();
+        } catch (Exception e) {
```

Comment:
We already called that method that BTW will never throw - can't we just return the currentBlobStore default name?